aws/sagemaker/large-model-inference/sample-llm/hf_acc_deploy_codellama_34b.ipynb

```diff
@@ -0,0 +1,398 @@
+    "print(f\"S3 Code or Model tar ball uploaded to --- > {code_artifact}\")\n",
+    "\n",
+    "model = Model(image_uri=image_uri, model_data=code_artifact, role=role)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "787cf1c3-ea9f-4a83-84d6-480bbb07f57b",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Step 4: Create SageMaker endpoint\n",
+    "You need to specify the instance to use and endpoint names"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08b153e2-3eb6-4882-aeb5-81d7efbb7a9a",
+   "metadata": {
+    "tags": [],
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "--------------"
+     ]
+    }
+   ],
+   "source": [
+    "instance_type = \"ml.g5.12xlarge\"\n",
+    "endpoint_name = sagemaker.utils.name_from_base(\"lmi-model-3-code-llama\")\n",
+    "\n",
+    "model.deploy(initial_instance_count=1,\n",
+    "             instance_type=instance_type,\n",
+    "             endpoint_name=endpoint_name,\n",
+    "             container_startup_health_check_timeout=3600\n",
+    "             )\n",
+    "\n",
+    "# our requests and responses will be in json format so we specify the serializer and the deserializer\n",
+    "predictor = sagemaker.Predictor(\n",
+    "    endpoint_name=endpoint_name,\n",
+    "    sagemaker_session=sess,\n",
+    "    serializer=serializers.JSONSerializer(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5e60c73-f3d8-43ef-9261-23677f03d5cb",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Step 5: Test and benchmark the inference\n",
+    "Firstly let's try to run with a wrong inputs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e21c0102-132c-4c18-8627-223592578c86",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "predictor.predict(\n",
+    "    {\"inputs\": \"def hello_world():\", \"parameters\": {\"max_new_tokens\":128, \"do_sample\":\"true\"}}\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa8fba6e-aa0b-44b1-82d5-9d369e38e8bb",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "790bf3f4-9398-41dd-8648-b6e43053b3e7",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "This can be done outside this notebook, in a bash shell terminal. The connection to the server is via the $SAGEMAKER url. The `awscurl` here is a benchmark tool, obtainable from \n",
+    "```\n",
+    "wget https://github.com/frankfliu/junkyard/releases/download/v0.2.2/awscurl && chmod +x awscurl\n",
```

Comment:
Since you already added, predict function, I dont think this awscurl command is needed. But I leave it up to you.